modules/t/Utils/FlatFile.t

```diff
@@ -24,15 +24,77 @@ use Test::Exception;
 use Bio::EnsEMBL::Compara::Utils::FlatFile;
 
 subtest 'map_row_to_header' => sub {
-    my $line = "a\tb\tc\td";
+    my $line_1 = "a\tb\tc\td";
+    my $line_2 = "a;b;c;d";
     my @header1 = (1, 2, 3, 4);
     my @header2 = (1, 2);
 
     is_deeply( 
-        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line, \@header1), 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1), 
         { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
     );
-    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line, \@header2) } qr/Number of columns in header do not match row/, "Header doesn't match";
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t3\t4"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+    
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1, "\t"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1, '\t'), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t3\t4", "\t"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header2) } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2")  } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    # to test LIMIT = -1 of split function: https://perldoc.perl.org/functions/split
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t")  } qr/Number of columns in header \(3\) do not match row \(4\)/, "Header doesn't match";
```

Comment:
I'm wondering, could this be inverted to check for the case where the data line is something like `"a\tb\tc\t"`? The reason I ask is that I think we'd be more likely to find an empty final column in a data line than in a header line.